### DISCUSSION TITLE
Is there a means for live reload to be triggered without an http (or similar) call

### DISCUSSION THREAD

**Discussion by lukens** [quarkusio/quarkus#31335] [Q&A]
If I'm working on developing a service where an http API isn't a core feature (if a feature at all), is there an alternative means to triggering a live-reload of code in dev mode?

For example, a small event-driven service for data-synchronising, where I don't want to bother calling every time I make a change, but would like to run the new code when next triggered by an event. Or, a service that uses `@Scheduled`, where I'd like the updated code to be run on the next schedule.

I feel like watching for file changes and doing a live reload on file saves would work fine for me, to avoid the need for any manual intervention to trigger the reload.

If not, would the be wider support for adding a parameter to enable reload on file save?

**Comment by maxandersen** [comment]
Quarkus extensions can add hooks wherever to trigger reloads. 

See https://quarkus.io/guides/writing-extensions#triggering-live-reload

**Reply by lukens** [reply]
Interesting, I'll have a look. 

I'm guessing the Scheduler extension doesn't implement these hooks.

**Reply by maxandersen** [reply]
Could be - sounds like a sensible enhancement! Want to give it a try?

**Reply by lukens** [reply]
I'd only been using Scheduler as a test case to try and see if things other than an http request would trigger reload, so I don't have an actual use case currently, otherwise I might have had a go at adding it (as it looks pretty simple).

I will maybe have a look if no-one else has, once I've got a bit more familiar with Quarkus.

**Reply by maxandersen** [reply]
Just in case you missed it a 's' keypress restart mnaually. If that helps.

**Reply by lukens** [reply]
I did see the `s` thing, however, that only seems to work with `dev` and not `remote-dev` (at least, when I try it)

**Reply by maxandersen** [reply]
Hmm...that should work. If not do please open bug.

**Reply by lukens** [reply]
Raised a bug - #31563
